### DISCUSSION TITLE
Database request from .liquid file

### DISCUSSION THREAD

**Discussion by c-sull** [kinngh/shopify-node-express-mongodb-app#50] [Q&A]
Hi everyone! I'm very new to this whole platform, and first of all wanted to say that this repo has been insanely helpful in the learning process! I'm working on an app that utilizes the database from both the frontend and backend, and have the frontend set up to use a .liquid file on a proxy in order to display data while keeping the original shop's template:

`proxyRouter.get("/", (req, res) => {
  res.set('Content-Type', 'application/liquid')
  .sendFile('server/routes/app_proxy/listings.liquid', { root: '.' })
});`
(in server/routes/index.js)

I was wondering if I have the right idea of just using javascript to make a request to an endpoint on the database? I see in the notes that you say to use TanStack or SWR for client requests, but as far as I'm aware this isn't possible when using a .liquid file. I guess my question is if I have the right idea on going about it? Or is there a way to display a .jsx file while still keeping the shop's original template? If it helps at all, I'm attempting to make an app that allows users to upload a few images and some other basic info from the proxy page, which gets put into the database and allows other users to see (like a buy/sell type thing), with the admin page allowing you to edit/remove listings. Not necessarily looking for a spoonfeed, I'm just wondering if I have the right idea. Any advice is appreciated!

**Comment by kinngh** [comment]
I'd make a theme extension instead that provides a section for the merchant to drop in. This would make the development process more streamlined, the URL would be of merchant's choosing instead of `https://store.com/apps/proxy-route/upload` and creating a custom UI in proxy is, IMO, cumbersome^

**Reply by c-sull** [reply]
That was my original plan, but I’m running into some weird issue with the theme I’m using. I’m developing this for a friends store as a hobby project, but they have a custom made theme installed. I originally made it include a theme app extension that added an app block, and it would display in dawn with no problem but when I changed it to a copy of their theme it didn’t show up anymore. I thought maybe the theme wasn’t made for 2.0, but if I change the target in schema to make it an app embed it shows up in the app embed section. I tried screwing around with modifying the theme to support app blocks but couldn’t get it to work and I didn’t want to worry about maintaining custom changes to a theme as I feel like that’s asking for issues down the road. 

Also, if I’m not mistaken isn’t an app block written in liquid as well? In that case, wouldn’t I still be running into the same issue of how to safely make requests? I understand it would streamline the process and make things more customizable, and I would love to use it if I could, but from what I understand I thought I’d still be having the same issue. Please correct me if I’m wrong though as you are obviously MUCH more experienced :)

**Reply by kinngh** [reply]
If that's the case, it's a problem with how their theme is engineered. Double check with them if blocks from other apps are showing up, if not, you want to check if you added the `presets` tag in your `schema` so it actually shows up as an option.

The app block is written in `liquid`, yes, but I'm pretty sure it's a minor error on either of your parts (less likely yours since it works in Dawn). You can double check your current code again by getting a theme from Shopify Theme Store on a trial and see if your app block(s) show up on that theme. If it does show up, it's a problem with the theme your friend has engineered and not your app, from what I can tell

Docs: https://shopify.dev/themes/architecture/sections/app-blocks
